tasks/configure.yml

```diff
@@ -0,0 +1,33 @@
+---
+- name: Create admins
+  become_user: www-data
+  shell: "wp user create {{ item.key }} {{ item.value.email }} --display_name='{{ item.value.name }}' --user_pass={{ item.value.password }} --role=administrator --path=/var/www/wordpress"
+  when: item.value.email is defined and item.value.password is defined
+  register: result
+  changed_when: "'Created user' in result.stdout"
+  failed_when: "result.rc != 0 and not 'username is already registered' in result.stderr"
+  with_dict: "{{ wordpress_admins }}"
+
+- name: Create admins
```

Comment:
You could combine both tasks each, and only add the password option if it is defined using inline if.